app/controllers/katello/api/v2/activation_keys_controller.rb

```diff
@@ -213,33 +213,28 @@ def remove_subscriptions
 
     api :PUT, "/activation_keys/:id/content_override", N_("Override content for activation_key")
     param :id, :number, :desc => N_("ID of the activation key"), :required => true
-    param :content_override, Hash, :desc => N_("Content override parameters"), :deprecated => true do
+    param :content_override, Hash, :desc => N_("Content override parameters") do
       param :content_label, String, :desc => N_("Label of the content"), :required => true
       param :value, String, :desc => N_("Override to a boolean value or 'default'"), :required => true
     end
-    param :content_overrides, Array, :desc => N_("Array of Content override parameters") do
+    param :content_overrides, Array, :desc => N_("Array of Content override parameters to be added in bulk") do
       param :content_label, String, :desc => N_("Label of the content"), :required => true
       param :value, String, :desc => N_("Override value. Provide a boolean value if name is 'enabled'"), :required => false
       param :name, String, :desc => N_("Override parameter key or name. Note if name is not provided the default name will be 'enabled'"), :required => false
       param :remove, :bool, :desc => N_("Set true to remove an override and reset it to 'default'"), :required => false
     end
     def content_override
       content_overrides = []
-      if params[:content_override]
-        ::Foreman::Deprecation.api_deprecation_warning("The parameter content_override will be removed in Katello 3.5. Please update to use the content_overrides parameter.")
-        content_override_params = {:content_label => params[:content_override][:content_label]}
-        value = params[:content_override][:value]
-        if value == "default"
-          content_override_params[:remove] = true
-        elsif value
-          content_override_params[:value] = ::Foreman::Cast.to_bool(value)
-        end
-        content_overrides << content_override_params
-      elsif params[:content_overrides]
-        content_overrides = params[:content_overrides]
-      end
+      content_overrides += [params[:content_override]] if params[:content_override]
+      content_overrides += params[:content_overrides] if params[:content_overrides]
+      organization = @activation_key.organization
+      all_content_labels = organization.content_labels
 
       content_override_values = content_overrides.map do |override_params|
+        unless override_params[:content_label] && all_content_labels.include?(override_params[:content_label])
```

Comment:
I think this check would not catch when someone sends a nil value for the content_label

Also, since we aren't ever using the values of `all_content_labels` it might be faster to check presence in the org like this:

```
unless organization.content.find_by_label(override_params[:content_label])

end
```
Another option would be to use a `where` and check the result size vs input size or something like that. That's probably better when sending a lot of overrides

Would it make sense to move this check inside of `validate_content_overrides_enabled` and address this problem in a few other places?

Reply:
good suggestions, I'll look into them :+1:

Reply:
The label check will work well (I didn't realize you can do `organization.contents`, thanks for pointing that out!)

For moving the check inside of `validate_content_overrides_enabled`, it does require an organization to be known, which doesn't seem as clear in other areas such as [host bulk actions](https://github.com/Katello/katello/blob/master/app/controllers/katello/api/v2/hosts_bulk_actions_controller.rb#L190-L206) so I think it can just stay here

Reply:
So how about:

```
labels = override_params.map { |override| override[:content_label] }
contents = organization.content.where(label: labels)

unless contents.size == labels.size # one more more weren't found
   missing_labels = contents.pluck(:label) - labels
   missing_labels.each |label|
     # error stuff
    end
end